### PR TITLE
Kdonga/vmware ps 2023

### PR DESCRIPTION
**Pull request by keyurdonga-msft**
## Description

Updated the AVS VMware cmdlets to the latest version [2023-03-01] from azure-rest-api-specs using autorest

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/VMware/VMware.Autorest/custom/Get-AzVMwareAddon.ps1 [hunk 1]
```diff
@@ -72,7 +72,7 @@ INPUTOBJECT <IVMwareIdentity>: Identity Parameter
 https://learn.microsoft.com/powershell/module/az.vmware/get-azvmwareaddon
 #>
 function Get-AzVMwareAddon {
-[OutputType([Microsoft.Azure.PowerShell.Cmdlets.VMware.Models.Api20211201.IAddon])]
+[OutputType([Microsoft.Azure.PowerShell.Cmdlets.VMware.Models.Api20230301.IAddon])]
```
Comment (wyunchi-ms) [inline src/VMware/VMware.Autorest/custom/Get-AzVMwareAddon.ps1:75]: The namespace doesn't contains API version any more.

**Comment by wyunchi-ms** [comment]
Get-ScriptCmdlet: Unable to find type [Microsoft.Azure.PowerShell.Cmdlets.VMware.Models.Api20230301.IAddon]

File: src/VMware/VMware.Autorest/custom/Get-AzVMwareAddon.ps1 (hunk 1 above)
Comment (keyurdonga-msft) [reply src/VMware/VMware.Autorest/custom/Get-AzVMwareAddon.ps1]: Fixed this in the latest build. It should not contain any api version.

**Comment by keyurdonga-msft** [comment]
All the tests have passed and recording files for all have been provided.

image

**Comment by wyunchi-ms** [comment]
Please fix the failed test cases and fill the examples

File: src/VMware/VMware.Autorest/test/AzVMwareAuthorization.Recording.json [hunk 2]
```diff
@@ -15,94 +15,94 @@
...
       "isContentBase64": false
     }
   },
-  "AzVMwareAuthorization+[NoContext]+List+$GET+https://management.azure.com/subscriptions/cef41485-ad1e-4cc3-a652-4c2620b8a2d0/resourceGroups/testgroup5q2kvd/providers/Microsoft.AVS/privateClouds/azps_test_cloudme6g/authorizations/yrles9?api-version=2021-12-01+2": {
+  "AzVMwareAuthorization+[NoContext]+List+$GET+https://localhost/subscriptions/cef41485-ad1e-4cc3-a652-4c2620b8a2d0/resourceGroups/testgroupp9uh7g/providers/Microsoft.AVS/privateClouds/azps_test_cloudb092/authorizations/authorization1?api-version=2023-03-01+2": {
     "Request": {
       "Method": "GET",
-      "RequestUri": "https://management.azure.com/subscriptions/cef41485-ad1e-4cc3-a652-4c2620b8a2d0/resourceGroups/testgroup5q2kvd/providers/Microsoft.AVS/privateClouds/azps_test_cloudme6g/authorizations/yrles9?api-version=2021-12-01",
+      "RequestUri": "https://localhost/subscriptions/cef41485-ad1e-4cc3-a652-4c2620b8a2d0/resourceGroups/testgroupp9uh7g/providers/Microsoft.AVS/privateClouds/azps_test_cloudb092/authorizations/authorization1?api-version=2023-03-01",
```
Comment (wyunchi-ms) [inline src/VMware/VMware.Autorest/test/AzVMwareAuthorization.Recording.json:32]: Where is the localhost come from?

Comment (AnaAriasRosales) [reply src/VMware/VMware.Autorest/test/AzVMwareAuthorization.Recording.json:32]: A mock api server was used to run the tests, which is configured to listen at localhost.

Comment (wyunchi-ms) [reply src/VMware/VMware.Autorest/test/AzVMwareAuthorization.Recording.json:32]: Then you can replace the `localhost` to `management.azure.com`. Our test framework doesn't support localhost yet

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**